app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      UserMailer.registration_confirmation(@user).deliver
+      flash[:notice] = "Thanks! Please check your email to complete sign up"
+      redirect_to sign_in_path
+    else
+      flash[:error] = "Uh oh. Something went wrong."
+      render "new"
+    end
+  end
+
+  def confirm_email
+    user = User.find_by_confirm_token(params[:id])
```

Comment:
This is awesome @npauzenga!!! :+1: I'm glad you are writing up your own authentication. 

I would recommend you move confirmation to its own controller. `Devise` follows a similar structure. That will make  your routes more "CRUD"/"RESTful". :smile_cat: .